### PR TITLE
refactor(ekf_localizer): add a function to erase a slash from the beginning of a string

### PR DESCRIPTION
**Pull request by IshitaTakeshi**
## Description

Add a function to erase a slash '/' from a beginning of a string 

## Tests performed

Check if the slash can be removed

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].
- [ ] The PR has been properly tested.
- [ ] The PR has been reviewed by the code owners.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.
- [ ] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: localization/ekf_localizer/src/ekf_localizer.cpp [hunk 1]
```diff
@@ -297,12 +298,9 @@ bool EKFLocalizer::getTransformFromTF(
   tf2::BufferCore tf_buffer;
   tf2_ros::TransformListener tf_listener(tf_buffer);
   rclcpp::sleep_for(std::chrono::milliseconds(100));
-  if (parent_frame.front() == '/') {
```
Comment (kminoda) [inline localization/ekf_localizer/src/ekf_localizer.cpp:300]: It seems a bit out of scope of this PR, but how about using this? https://github.com/autowarefoundation/autoware.universe/blob/main/common/tier4_autoware_utils/include/tier4_autoware_utils/ros/transform_listener.hpp

This getTransformFromTF function is a legacy function according to @.YamatoAndo -san, so I think we can completely replace it with the above library class.

Comment (IshitaTakeshi) [reply localization/ekf_localizer/src/ekf_localizer.cpp:300]: If we don't repeat calling lookupTransform, or if we just show the warning message and do nothing more than this, I can use the proposed function.

Comment (kminoda) [reply localization/ekf_localizer/src/ekf_localizer.cpp:300]: @IshitaTakeshi The `EKFLocalizer::getTransformFromTF` function has some tricky code such as `rclcpp::sleep_for(std::chrono::milliseconds(100));`, so I think using the one from tier4_autoware_utils would make much more sense than leaving this function as it is. Given that this tier4_autoware_utils::transform_listener is already used in several packages, I think it is OK to replace it. (not necessarily in this PR, though)

Comment (IshitaTakeshi) [reply localization/ekf_localizer/src/ekf_localizer.cpp:300]: OK then I'm going to do it or continue the discussion on another PR so could you approve this PR first?

File: localization/ekf_localizer/include/ekf_localizer/string.hpp [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef EKF_LOCALIZER__STRING_HPP_
+#define EKF_LOCALIZER__STRING_HPP_
+
+#include <string>
+
+inline std::string eraseBeginSlash(const std::string & s)
```
Comment (kminoda) [inline localization/ekf_localizer/include/ekf_localizer/string.hpp:20]: How about `eraseLeadingSlash` or `eraseLeadingForwardSlash`?

**Review by kminoda — APPROVED** [review]

File: localization/ekf_localizer/src/ekf_localizer.cpp (hunk 1 above)
Comment (kminoda) [reply localization/ekf_localizer/src/ekf_localizer.cpp:300]: OK :+1:

File: localization/ekf_localizer/include/ekf_localizer/string.hpp (hunk 2 above)
Comment (kminoda) [reply localization/ekf_localizer/include/ekf_localizer/string.hpp:20]: (Sorry I approved mistakenly, so dismissed it)

Comment (IshitaTakeshi) [reply localization/ekf_localizer/include/ekf_localizer/string.hpp]: Updated in [e95c689](https://github.com/autowarefoundation/autoware.universe/pull/2702/commits/e95c68914665df278eda5eddfce8904383789f4b)

**Review by kminoda — APPROVED** [review]
LGTM

**Pull request merged**